README.md

```diff
@@ -1,56 +1,261 @@
+        printf("Found match: '%.*s'\n", (int)(ovector[1] - ovector[0]),
+               subject + ovector[0]);
+    }
+
+    pcre2_match_data_free(match_data);   /* Free resources */
+    pcre2_code_free(re);
+    return 0;
+}
+----------------------------------------------------------------------
+
+# Compile the demo:
+gcc -g -I./pcre2/build -L./pcre2/build demo.c -o demo -lpcre2-8
+
+# Finally, run our demo:
+./demo 'c.t' 'dogs and cats'
+
+# We fetched, built, and called PCRE2 successfully! :)
+```
+
+</details>
+
+---
+
+The main ways of obtaining PCRE2 are:
+
+1. Via Git clone:
+
+    ```
     git clone https://github.com/PCRE2Project/pcre2.git
-    svn co    https://github.com/PCRE2Project/pcre2.git
+    ```
+
+    Please use a release tag in production, not the development branch!
+
+2. Via download of the [release tarball](https://github.com/PCRE2Project/pcre2/releases/latest).
+
+3. Finally, PCRE2 is also bundled by various downstream package managers (such as Linux distributions, or [vcpkg](https://vcpkg.io/)). These are provided by third parties, not the PCRE2 project.
+
+The main ways of building PCRE2 are:
+
+1. Via CMake (Linux/Windows/macOS, and others)
+
+    ```
+    cd pcre2/
+    cmake -B build .
+    cmake --build build/
+    ```
+
+2. Via Autoconf (Linux/Unix)
+
+    ```
+    cd pcre2/
+    ./configure
+    make
+    ```
+
+See ["Platforms"](#platforms) below for links to more detailed build documentation.
+
+## API Overview
+
+The PCRE2 API supports strings in 8-bit, 16-bit, and 32-bit encodings, with or without UTF encoding. There is also EBCDIC support.
+
+The default regular expression dialect closely matches the syntax and behaviour of Perl 5, with PCRE2-specific extensions. A wide variety of granular flags can be passed to the PCRE2 API to customise this to more closely follow other dialects such as JavaScript or Python.
+
+The default matching engine uses a depth-first tree search with backtracking, which is highly feature-rich but has worst-case exponential time (PCRE2 allows aborting the match if a time limit is exceeded, expressed as a maximum number of steps in the tree search). The second matching engine uses a JIT for greatly improved performance, compiling the regular expression to a block of equivalent native machine code.
+
+PCRE2 has a third matching engine, using a DFA engine which is generally slower, but has worst-case polynomial matching time and is able to find the POSIX-style "leftmost-longest" match.
+
+There are accompanying utility functions for converting glob patterns and POSIX BRE/ERE patterns to PCRE2 regular expressions; and also for performing high-level regular expression operations such as search-and-replace with a powerful replacement string syntax.
+
+As well as the PCRE2 API, the library also offers a POSIX-compatible `<regex.h>` header and `regexec()` function. However, this does not provide the ability to pass PCRE2 flags, so we recommend users consume the PCRE2 API if possible.
```

Comment:
pcre2posix.h is actually no longer "POSIX" compatible, because it only supports "int" sized offsets

Reply:
Well yes, but even glibc has the same problem in their regex.h (_REGEX_LARGE_OFFSETS is not POSIX compliant). The POSIX guys just randomly changed it...

Whenever they fix it so that snprintf returns a size_t, we can worry about regoff_t. There's a vast number of standard C and POSIX C interfaces that incorrectly specify int, and the regex functions are the least of everyone's worries.

We should just ignore the problem, since we didn't start it.